File: src/main/java/pl/training/workshop/m6/s07_decorator/step3/Glasses3D.java
package pl.training.workshop.m6.s07_decorator.step3;

import pl.training.workshop.shared.Money;

/** Krok 3: okulary 3D jako dekorator; decyzja "czy potrzebne" została w fabryce. */
public record Glasses3D(PricedTicket inner) implements PricedTicket {
    @Override
    public Money price() {
        return inner.price().plus(Money.of("3.00"));
    }

    @Override
    public String description() {
        return inner.description() + " +okulary 3D";
    }
}

File: csharp/src/Training.Workshop/M6/S07Decorator/Step3/Glasses3D.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S07Decorator.Step3;

/// <summary>
/// Krok 3: okulary 3D jako dekorator; decyzja "czy potrzebne" została w fabryce.
/// </summary>
public sealed record Glasses3D(IPricedTicket Inner) : IPricedTicket
{
    public Money Price()
    {
        return Inner.Price().Plus(Money.Of("3.00"));
    }

    public string Description()
    {
        return Inner.Description() + " +okulary 3D";
    }
}
